src/DeviceStatusPage/GroupsCard.tsx

```diff
@@ -59,24 +67,31 @@ const GroupsCard = ({
   device,
   showAddToGropButton,
   onAddToGroupClick,
-}: GroupsCardProps): React.ReactElement => (
-  <FullHeightCard xs={12} md={6} className="mb-4">
-    <Card.Header as="h5">Groups</Card.Header>
-    <Card.Body className="d-flex flex-column">
-      {device.groups.length > 0 ? (
-        <GroupsTable groups={device.groups} />
-      ) : (
-        <p>Device does not belong to any group</p>
-      )}
-      <div className="mt-auto">
-        {showAddToGropButton && (
-          <Button variant="primary" onClick={onAddToGroupClick}>
-            Add to existing group
-          </Button>
+}: GroupsCardProps): React.ReactElement => {
+  const astarte = useAstarte();
+  return (
+    <FullHeightCard xs={12} md={6} className="mb-4">
+      <Card.Header as="h5">Groups</Card.Header>
+      <Card.Body className="d-flex flex-column">
+        {device.groups.length > 0 ? (
+          <GroupsTable groups={device.groups} />
+        ) : (
+          <p>Device does not belong to any group</p>
         )}
-      </div>
-    </Card.Body>
-  </FullHeightCard>
-);
+        <div className="mt-auto">
+          {showAddToGropButton && (
+            <Button
+              variant="primary"
+              onClick={onAddToGroupClick}
+              hidden={!astarte.token?.can('appEngine', 'PATCH', `/devices/${device.id}`)}
```

Comment:
Here we already have a `showAddToGroupButton` variable.
So instead of editing this component, we can modify the value set for this variable by the parent component.

Currently, the prop is set by the parent component as:
```tsx
showAddToGropButton={unjoinedGroups.length > 0}
```
But we could instead also require that the token is able to modify at least one of the `unjoinedGroups`.

```tsx
const canAddDeviceToGroups = unjoinedGroups.some(groupName => astarte.token?.can('appEngine', 'POST', `/groups/${'groupName'}/devices`));

...

<GroupsCard
  showAddToGropButton={canAddDeviceToGroups}
  ...
/>
```